src/App.vue

```diff
@@ -15,21 +22,139 @@ export default {
 
   data() {
     return {
-      messages: []
+      intervalId: null,
+      isLoading: false,
+      messages: [],
+      refreshInterval: 30
     };
   },
 
   methods: {
+    startUpdateMessages() {
+      this.intervalId = setInterval(() => {
+        this.updateMessages();
+      }, this.refreshInterval * 1000);
+    },
     updateMessages() {
-      api.getMessages().then(messages => (this.messages = messages));
+      this.isLoading = true;
+      this.messages = [];
+      api.getMessages().then(messages => {
+        this.messages = messages;
+        this.isLoading = false;
+      });
+    },
+    updateRefreshInterval(newRefreshInterval) {
+      this.refreshInterval = newRefreshInterval;
+      clearInterval(this.intervalId);
+      if (this.refreshInterval) this.startUpdateMessages();
+      else this.isLoading = false;
     }
   },
 
   created() {
     this.updateMessages();
-    setInterval(() => {
-      this.updateMessages();
-    }, 10000);
+    this.startUpdateMessages();
+  },
+
+  beforeDestroy() {
+    clearInterval(this.intervalId);
   }
 };
 </script>
+
+<style scoped>
+.loader {
```

Comment:
Could you put this is a css file alone, It make things more complex to read here.